lib/bamboo/adapters/sendgrid_helper.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Bamboo.SendgridHelper do
+  @moduledoc """
+  Functions for using features specific to Sendgrid template substitution tags.
+
+  Only one template can be specified, but multiple tags can be added to the
+  email. Ordering of the function calls do not make a difference.
+
+  ## Example
+
+      email
+      |> with_template("80509523-83de-42b6-a2bf-54b7513bd2aa")
+      |> substitute("%name%", "Jon Snow")
+      |> substitute("%location%", "Westeros")
+  """
+
+  alias Bamboo.Email
+
+  @id_size 36
+
+  @doc """
+  Specify the template for SendGrid to use for the substitutions
+
+  The `template_id` must be a valid `UUID` as generated for each template by
+  SendGrid. If called multiple times, the last template specified will be used.
```

Comment:
I was just attempting to err on the side of being explicit, that if someone is building up a message progressively across multiple functions, the last template specified would be used. This is probably obvious in intent and could be removed.